File: _site_local/mod_10_paradigmas_e_padroes_de_projeto/exemplos/oo/Kotlin/Retangulo/src/Retangulo.kt
// Retangulo.kt
class Retangulo(val largura: Double, val altura: Double) {

    // Método para calcular a área do retângulo
    fun calcularArea(): Double {
        return largura * altura
    }

    // Método para exibir as informações do retângulo
    fun exibirInformacoes() {
        println("Largura: $largura metros")
        println("Altura: $altura metros")
        println("Área: ${calcularArea()} metros quadrados")
    }

    // Método para comparar a área com outro retângulo
    fun compararArea(outroRetangulo: Retangulo) {
        val areaAtual = this.calcularArea()
        val areaOutro = outroRetangulo.calcularArea()

        if (areaAtual > areaOutro) {
            println("O Retângulo X (Área: $areaAtual) tem uma área maior que o Retângulo Y (Área: $areaOutro).")
        } else if (areaAtual < areaOutro) {
            println("O Retângulo Y (Área: $areaOutro) tem uma área maior que o Retângulo X (Área: $areaAtual).")
        } else {
            println("O Retângulo X (Área: $areaAtual) e o Retângulo Y (Área: $areaOutro) têm a mesma área.")
        }
    }
}

File: _site_local/mod_10_paradigmas_e_padroes_de_projeto/exemplos/oo/JavaScript/Retangulo/Retangulo.js
// Classe Retangulo
class Retangulo {
    constructor(largura, altura) {
        this.largura = largura;
        this.altura = altura;
    }

    // Método para calcular a área do retângulo
    calcularArea() {
        return this.largura * this.altura;
    }

    // Método para exibir as informações do retângulo
    exibirInformacoes() {
        console.log(`Largura: ${this.largura} m`);
        console.log(`Altura: ${this.altura} m`);
        console.log(`Área: ${this.calcularArea()} m²`);
    }

    // Método para comparar a área com outro retângulo
    compararArea(outroRetangulo) {
        const areaAtual = this.calcularArea();
        const areaOutro = outroRetangulo.calcularArea();

        if (areaAtual > areaOutro) {
            console.log(`O Retângulo X (Área: ${areaAtual} m²) tem uma área maior que o Retângulo Y (Área: ${areaOutro} m²).`);
        } else if (areaAtual < areaOutro) {
            console.log(`O Retângulo Y (Área: ${areaOutro} m²) tem uma área maior que o Retângulo X (Área: ${areaAtual} m²).`);
        } else {
            console.log(`O Retângulo X (Área: ${areaAtual} m²) e o Retângulo Y (Área: ${areaOutro} m²) têm a mesma área.`);
        }
    }
}

// Exporta a classe para ser usada em outro arquivo
module.exports = Retangulo;
